static/js/typeahead_helper.js

```diff
@@ -207,17 +207,38 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+function activity_score(stream) {
+    var stream_score = 0;
+    if (stream.pin_to_top) {
+        stream_score += 2;
+    }
+    if (stream_data.is_active(stream.name)) {
+        stream_score += 1;
+    }
+    return stream_score;
+}
+
+// Find which stream comes first in the sidebar listing order
+exports.compare_by_sidebar = function (stream_a, stream_b) {
+    var a_score = activity_score(stream_a);
+    var b_score = activity_score(stream_b);
+    if (a_score === b_score) {
```

Comment:
This code is fine, but it's usually a good convention in these compare functions to early-return when the first criteria are unequal.

~~~ js
var diff = a_score - b_score

if (diff !== 0) {
    return diff;
}

return exports.sub_by_count(stream_a, stream_b);